src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,375 @@
+
+    Observable
+      .fromEvent(document, 'mouseup')
+      .takeWhile(() => {
+        return this.isDragging;
+      })
+      .subscribe((mouseUpEvent: any) => {
+        this.onHandleRelease(mouseUpEvent);
+      });
+  }
+
+  public onMouseMove(event: MouseEvent): void {
+    /* Sanity check */
+    /* istanbul ignore if */
+    if (!this.isDragging) {
+      return;
+    }
+
+    const offsetX = event.clientX - this.xCoord;
+    let width = this.drawerWidth;
+
+    width += offsetX;
+
+    if (width < this.drawerWidthMin || width > this.drawerWidthMax) {
+      return;
+    }
+
+    this.drawerWidth = width;
+
+    this.xCoord = event.clientX;
+    this.changeDetectorRef.markForCheck();
+  }
+
+  public onHandleRelease(event: MouseEvent): void {
+    this.isDragging = false;
+    this.coreAdapterService.toggleIframePointerEvents(true);
+    this.changeDetectorRef.markForCheck();
+  }
+
+  public onResizeHandleChange(event: any): void {
+    this.drawerWidth = event.target.value;
+    this.setDrawerMaxWidth();
+  }
+
+  public onShowDrawerButtonClick() {
+    /* istanbul ignore else */
+    if (this.beforeWorkspaceClose.observers.length === 0) {
+      this.isDrawerVisible = true;
+    } else {
+      this.beforeWorkspaceClose.emit(new SkySplitViewBeforeWorkspaceCloseHandler(() => {
+        this.isDrawerVisible = true;
+        this.changeDetectorRef.markForCheck();
+      }));
+    }
+  }
+
+  @HostListener('window:resize', ['$event'])
+  public onWindowResize(event: any): void {
+    // If window size is smaller than drawerWidth + tolerance, shrink drawerWidth.
+    if (this.isDrawerVisible && event.target.innerWidth < this.drawerWidth + this.widthTolerance) {
+      this.drawerWidth = event.target.innerWidth - this.widthTolerance;
+    }
+    this.updateBreakpoints();
+  }
+
+  public onWorkspaceEnterComplete(): void {
+    this.animationComplete.next();
+  }
+
+  private setDrawerMaxWidth(): void {
+    this.drawerWidthMax = this.skyWindow.nativeWindow.innerWidth - this.widthTolerance;
+    setTimeout(() => {
+      this.changeDetectorRef.markForCheck();
+    });
+  }
+
+  private handleIncomingMessages(message: SkySplitViewMessage) {
+    /* tslint:disable-next-line:switch-default */
+    switch (message.type) {
+      case SkySplitViewMessageType.FocusWorkspace:
+        // If mobile, wait until animation is complete then set focus on workspace panel.
+        // Otherwise, just set focus right away.
+        if (!this.workspaceVisible) {
+          this.isDrawerVisible = false;
+          this.animationComplete
+            .take(1)
+            .subscribe(() => {
+              this.applyAutofocus();
+            });
+        } else {
+          this.applyAutofocus();
+        }
+        this.changeDetectorRef.markForCheck();
+        break;
+    }
+  }
+
+  private updateBreakpoints(): void {
+    // Update drawer component.
+    if (this.drawerComponent) {
+      this.drawerComponent.updateBreakpoint(this.drawerWidth);
```

Comment:
Since we already have access to the elements wrapping the drawer and workspace components, telling each component to resize itself feels like a scope breach. I would either:
1. Move the template and width logic to the drawer and workspace components. They would listen for breakpoint changes themselves and resize accordingly.
2. Remove the `updateBreakpoint` methods from the drawer and workspace components and set the width of the wrapping elements in the split-view component directly.

Reply:
This is also related to my attempt to put `width` on the drawer component. I'm already heading down this path, and I'll update this PR with that work soon.

Reply:
This has been restructured - The components no longer tell the children what to do - instead they just communicate with the `SplitViewService`.